misc/metrics/src/relay.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 Protocol Labs.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use open_metrics_client::encoding::text::Encode;
+use open_metrics_client::metrics::counter::Counter;
+use open_metrics_client::metrics::family::Family;
+use open_metrics_client::registry::Registry;
+
+pub struct Metrics {
+    events: Family<EventLabels, Counter>,
+}
+
+impl Metrics {
+    pub fn new(registry: &mut Registry) -> Self {
+        let sub_registry = registry.sub_registry_with_prefix("relay");
+
+        let events = Family::default();
+        sub_registry.register(
+            "events",
+            "Events emitted by the relay NetworkBehaviour",
+            Box::new(events.clone()),
+        );
+
+        Self { events }
+    }
+}
+
+#[derive(Clone, Hash, PartialEq, Eq, Encode)]
+struct EventLabels {
+    event: EventType,
+}
+
+#[derive(Clone, Hash, PartialEq, Eq, Encode)]
```

Comment:
```suggestion
#[derive(Debug, Clone, Hash, PartialEq, Eq, Encode)]
struct EventLabels {
    event: EventType,
}

#[derive(Debug, Clone, Hash, PartialEq, Eq, Encode)]
```

Reply:
:+1:   Fixed in a9d9b4158462a7b1f1d285a46d0de730b03e7754.